boards/arm/stm32f746g_disco/stm32f746g_disco.dts

```diff
@@ -119,3 +120,67 @@
 		     &eth_txd0_pg13
 		     &eth_txd1_pg14>;
 };
+
+&flash0 {
```

Comment:
flash layout changes are not related to QSPI, so they should go into a separate commit (or even a new PR if not related/required with these changes)